core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -60,6 +60,34 @@ object ConfigSource {
       SystemEnvironment :: Nil
     )
 
+  /** Forming configuration from command line arguments, eg `List(--param1=xxxx, --param2=yyyy)` */
+  def fromArgs(args: List[String], separator: Option[String] = None): ConfigSource[String, String] = {
+    val map =
+      args
+        .filter(s => s.startsWith("--") && s.contains("="))
```

Comment:
This ignores any args that aren't switches, ie of the form "--param=value".
The idea was to allow command line switches to be processed while ignoring other command args.
An alternative approach would be to treat them as an error.
What do you think?

Reply:
I think we should take a different approach here: pack all of the command-line arguments into multiple property lists.

Using `PropertyTree.mergeAll`, we can merge a bunch of command line options into the smallest possible set of property trees, and then use those property trees to perform lookup.

All the hard logic is embedded inside `mergeAll` / `merge`, so if the command-line options form multiple conflicting schemas, then they'll be broken apart into multiple property trees.

Just loop through all command line arguments, looking for the form:

```
--xyz=xyz
--xyz xyz
-xyz=xyz
-xyz xyz
```

Using `.` as the path separator for the keys.

Extract each pair as a separate `PropertyTree`. This will give you a `List`, which you can call `mergeAll` on to "compact" it into compatible schemas.

Probably, you should sort the list by order of decreasing size. This will ensure that in the event of conflicts or unrelated command line options, you get the "biggest" merge.

Then you can construct a `ConfigSource` that chooses between the different `PropertyTree`, starting from the biggest and proceeding to the smallest.

Finally, I'd add a `Config.fromArgs` helper method that does what all the other helper methods do there.